File: src/main/java/com/lyaiya/softcode/mixin/blockdrops/MixinCategory.java
package com.lyaiya.softcode.mixin.blockdrops;

import com.lyaiya.softcode.constant.KeyConstant;
import com.lyaiya.softcode.constant.ModIdConstant;
import com.lyaiya.softcode.util.TranslationKeyKt;
import mrriegel.blockdrops.Category;
import net.minecraft.client.resources.I18n;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;

@Mixin(
        value = Category.class,
        remap = false
)
abstract class MixinCategory {
    @Inject(
            method = "getTitle",
            at = @At("RETURN"),
            cancellable = true
    )
    private void injectGetTitle(CallbackInfoReturnable<String> cir) {
        String key = TranslationKeyKt.create(ModIdConstant.BLOCK_DROPS, KeyConstant.GUI, KeyConstant.RECIPE_CATEGORY, "block_drops");
        cir.setReturnValue(I18n.format(key));
    }
}

File: src/main/java/com/lyaiya/softcode/mixin/blockdrops/MixinCategory.kt
package com.lyaiya.softcode.mixin.blockdrops

import com.lyaiya.softcode.constant.*
import com.lyaiya.softcode.util.*
import mrriegel.blockdrops.*
import net.minecraft.client.resources.*
import org.spongepowered.asm.mixin.*
import org.spongepowered.asm.mixin.injection.*
import org.spongepowered.asm.mixin.injection.callback.*

@Mixin(Category::class, remap = false)
abstract class MixinCategory {
    @Inject(
        method = ["getTitle"],
        at = [At("RETURN")],
        cancellable = true
    )
    private fun injectGetTitle(cir: CallbackInfoReturnable<String>) {
        cir.returnValue = I18n.format(createTranslationKey(ModIdConstant.BLOCK_DROPS, KeyConstant.GUI, KeyConstant.RECIPE_CATEGORY, "block_drops"))
    }
}